### ISSUE TITLE
Incorrect sorting with mixed Y, YMD, and YMD-YMD dates

### ISSUE THREAD

**Issue by njbart** [jgm/pandoc-citeproc#416]
Sorting by date is incorrect when dates with year precision are mixed with dates with month or day precision, or with date ranges.

(I stumbled across this when using a modified `chicago-author-date.csl` set up to sort the references section entries chronologically, but the effect is demonstrated just as clearly with the standard `chicago-author-date.csl` if all contributors are identical, as in the following example. ` type: article-newspaper` is merely used to have the complete dates appear in the bibliography output; same effect with other types, e.g., `book`.)

```
pandoc -F pandoc-citeproc -t plain  << EOT

# Text

Blah blah [@item1; @item2; @item3].

# References {-}

---
title: The Title
references:
- id: item1
  type: article-newspaper
  author:
  - family: Doe
    given: J.
  issued:
  - year: 2010
    month: 12
    day: 13
  title: The title
- id: item2
  type: article-newspaper
  author:
  - family: Doe
    given: J.
  issued:
  - year: 2007
    month: 12
    day: 12
  - year: 2007
    month: 12
    day: 13
  title: The title
- id: item3
  type: article-newspaper
  author:
  - family: Doe
    given: J.
  issued:
  - year: 2008
  title: The title
...

EOT
```

Actual output:

```
[...]


Doe, J. 2008. “The Title,” 2008.

———. 2010. “The Title,” December 13, 2010.

———. 2007. “The Title,” December 12–13, 2007.
```

Expected output:

```
[...]


Doe, J. 2007. “The Title,” December 12–13, 2007.

———. 2008. “The Title,” 2008.

———. 2010. “The Title,” December 13, 2010.

```

**Comment by tzok** [comment]
I have just stumbled upon this same bug. Below I put a minimal working example that anyone can easily try to confirm there is a problem.


Here is a minimal CSL JSON: (take note of the `date-parts` with 3 components for `ref-1` and 2 components for `ref-2`)

```json
[
    {
        "id": "ref-1",
        "type": "article-journal",
        "title": "Title 1",
        "issued": {
            "date-parts": [
                [
                    "2010",
                    1,
                    1
                ]
            ]
        }
    },
    {
        "id": "ref-2",
        "type": "article-journal",
        "title": "Title 2",
        "issued": {
            "date-parts": [
                [
                    "2020",
                    1
                ]
            ]
        }
    }
]
```

Here is an excerpt from CSL:

```xml
<bibliography>
  <sort>
    <key variable="issued" sort="descending"/>
  </sort>
</bibliography>
```

And here is the minimal Markdown file:

```markdown
---
nocite: '@*'
---
```

Expected outcome:

```html
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-ref-2">
<p><em>Title 2</em>. 2020. </p>
</div>
<div id="ref-ref-1">
<p><em>Title 1</em>. 2010. </p>
</div>
</div>
```

Current outcome:

```html
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-ref-1">
<p><em>Title 1</em>. 2010. </p>
</div>
<div id="ref-ref-2">
<p><em>Title 2</em>. 2020. </p>
</div>
</div>
```

Workaround: to add day component of date in `issued` of `ref-2`. The day is not displayed by the CSL anyway.

**Comment by jgm** [comment]
This will be fixed in an upcoming version of pandoc incorporating jgm/citeproc.
Test has already been added.

**Comment by njbart** [comment]
This has been fixed in pandoc 2.11. Closing.
